boards/hifive_inventor/src/main.rs

```diff
@@ -0,0 +1,299 @@
+    );
+    DynamicDeferredCall::set_global_instance(dynamic_deferred_caller);
+
+    // Configure kernel debug gpios as early as possible
+    kernel::debug::assign_gpios(None, None, None);
+
+    // Create a shared UART channel for the console and for kernel debug.
+    let uart_mux = components::console::UartMuxComponent::new(
+        &peripherals.e310x.uart0,
+        115200,
+        dynamic_deferred_caller,
+    )
+    .finalize(());
+
+    let hardware_timer = static_init!(
+        sifive::clint::Clint,
+        sifive::clint::Clint::new(&e310_g003::clint::CLINT_BASE)
+    );
+
+    // Create a shared virtualization mux layer on top of a single hardware
+    // alarm.
+    let mux_alarm = static_init!(
+        MuxAlarm<'static, sifive::clint::Clint>,
+        MuxAlarm::new(hardware_timer)
+    );
+    hil::time::Alarm::set_alarm_client(hardware_timer, mux_alarm);
+
+    // Alarm
+    let virtual_alarm_user = static_init!(
+        VirtualMuxAlarm<'static, sifive::clint::Clint>,
+        VirtualMuxAlarm::new(mux_alarm)
+    );
+    virtual_alarm_user.setup();
+
+    let systick_virtual_alarm = static_init!(
+        VirtualMuxAlarm<'static, sifive::clint::Clint>,
+        VirtualMuxAlarm::new(mux_alarm)
+    );
+    systick_virtual_alarm.setup();
+
+    let alarm = static_init!(
+        capsules::alarm::AlarmDriver<'static, VirtualMuxAlarm<'static, sifive::clint::Clint>>,
+        capsules::alarm::AlarmDriver::new(
+            virtual_alarm_user,
+            board_kernel.create_grant(capsules::alarm::DRIVER_NUM, &memory_allocation_cap)
+        )
+    );
+    hil::time::Alarm::set_alarm_client(virtual_alarm_user, alarm);
+
+    let chip = static_init!(
+        e310_g003::chip::E310x<E310G003DefaultPeripherals>,
+        e310_g003::chip::E310x::new(peripherals, hardware_timer)
+    );
+    CHIP = Some(chip);
+
+    let process_printer =
+        components::process_printer::ProcessPrinterTextComponent::new().finalize(());
+    PROCESS_PRINTER = Some(process_printer);
+
+    // Need to enable all interrupts for Tock Kernel
+    chip.enable_plic_interrupts();
+
+    // enable interrupts globally
+    csr::CSR
+        .mie
+        .modify(csr::mie::mie::mext::SET + csr::mie::mie::msoft::SET + csr::mie::mie::mtimer::SET);
+    csr::CSR.mstatus.modify(csr::mstatus::mstatus::mie::SET);
+
+    // Setup the console.
+    let console = components::console::ConsoleComponent::new(
+        board_kernel,
+        capsules::console::DRIVER_NUM,
+        uart_mux,
+    )
+    .finalize(components::console_component_helper!());
+    // Create the debugger object that handles calls to `debug!()`.
+    components::debug_writer::DebugWriterComponent::new(uart_mux).finalize(());
+
+    let lldb = components::lldb::LowLevelDebugComponent::new(
+        board_kernel,
+        capsules::low_level_debug::DRIVER_NUM,
+        uart_mux,
+    )
+    .finalize(());
+
+    // Need two debug!() calls to actually test with QEMU. QEMU seems to have
```

Comment:
Do you mean to get some output in qemu, you must repeat the `debug!()` every time?  For instance,
```
debug!("Hello");
debug!("Hello");
>>
Hello
```

Reply:
I don't really know what that comment really means. I borrowed it from the [hifive1 board source](https://github.com/tock/tock/blame/93d50be711b959d5dceeb17f8d4b60c6cee3e2b3/boards/hifive1/src/main.rs#L254).
But given that there is no QEMU support for this board, I will consider removing it. It was helpful though when debugging to have two debug messages. I had some problems when only one of them was printed, but they are solved now.

Reply:
My understanding is that a single `debug!` works? If so, please remove the comment or says soemthing like *if you plan using this throgh QEMU, please consider using two debug! statements as this seems to be a QEMU issue*.